### DISCUSSION TITLE
Does DETECT_EXTERNAL_IP=yes run the script only on startup or periodically?

### DISCUSSION THREAD

**Discussion by kladderadeng** [coturn/coturn#1479] [Q&A]
I am using coturn/coturn:latest for my coturn container and because I am behind a NAT with a dynamic IP, I use DETECT_EXTERNAL_IP=yes in my compose file. I wonder if the script to detect the external IP is only run once on startup or if it is triggered periodically to adjust the coturn configuration when the external IP changed. Does someone know how it is implemented?

**Comment by tangxinfa** [comment]
Why i can't found any reference of DETECT_EXTERNAL_IP in the code base?  How the "DETECT_EXTERNAL_IP=yes" magic works?
